### PR TITLE
[OPS-851] Add update-niv.sh script

### PR DESCRIPTION
**Pull request by balsoft**
Adds a script that can be used to automatically update dependencies via niv.

### CODE REVIEW

File: scripts/update-niv.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+changes() {
+    diff -y --suppress-common-lines <(echo "$1") <(echo "$2") | awk '{print $1": "substr($2, 0, 7)"->"substr($5, 0, 7)}' | column -t
+}
+
+new_pr() {
+    hub pull-request --no-edit -r serokell/operations -m "Update dependencies [automatic]" -m "$1"
+}
+[ -d nix ]
+
+git checkout -B "automatic/update"
```
Comment (mkaito) [inline scripts/update-niv.sh:28]: Add something like `-$(date +%s)` here or something to avoid collisions.

File: scripts/update-niv.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+[ $(git diff | wc -l) -eq 0 ] || {
+    git add nix
+    git commit --author=$author -m "Update dependencies with niv [automatic]" -m "$(changes "$before" "$after")"
+}
+
+
+[ $(git diff origin/master | wc -l) -eq 0 ] || {
+    git push --set-upstream origin automatic/update
+    hub pr list | grep "Update dependencies \[automatic\]" || new_pr "\`\`\`$(changes "$before" "$after")\`\`\`"
```
Comment (mkaito) [inline scripts/update-niv.sh:54]: It's ironic that you change for duplicate PR but not for duplicate branch :joy:

File: scripts/update-niv.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -p git jq gitAndTools.hub niv -i bash
+
+set -euo pipefail
+
+author="serokell-bot <operations+github@serokell.io>"
+
+pkgs() {
+    jq ".[] | .repo, .rev" nix/sources.json | tr -d \" | xargs -n2 echo
```
Comment (mkaito) [inline scripts/update-niv.sh:10]: I really like documenting these opaque chains of commands for future reference

File: scripts/update-niv.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+changes() {
+    diff -y --suppress-common-lines <(echo "$1") <(echo "$2") | awk '{print $1": "substr($2, 0, 7)"->"substr($5, 0, 7)}' | column -t
```
Comment (mkaito) [inline scripts/update-niv.sh]: Explain what's going on here, please

**Review by mkaito — CHANGES_REQUESTED** [review]

File: scripts/update-niv.sh (hunk 1 above)
Comment (mkaito) [reply scripts/update-niv.sh:28]: Nevermind, missed your `push --force` below.

File: scripts/update-niv.sh (hunk 4 above)
Comment (balsoft) [reply scripts/update-niv.sh]: Explained (probably) :smile:

File: scripts/update-niv.sh (hunk 2 above)
Comment (balsoft) [reply scripts/update-niv.sh:54]: Added `--force` (it was intended, I just sort of forgot about it)

File: scripts/update-niv.sh (hunk 3 above)
Comment (balsoft) [reply scripts/update-niv.sh:10]: Writing them is much easier than reading them.

**Comment by mkaito** [comment]
So what happens here is that you run this script on a timer and if there's something to update, it does a PR, or updates an existing one if we were lazy and didn't merge it, right?

**Review by mkaito — APPROVED** [review]
Looks good now

**Comment by balsoft** [comment]
Yes. Should I add that to the documentation?

**Pull request merged**